torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -527,7 +527,11 @@ class ShapePropagator {
         }
         return;
       }
-      case aten::tensor: {
+      case aten::tensor:
+      case aten::as_tensor: {
+        if (node->inputs().at(0)->type()->isSubtypeOf(TensorType::get())) {
```

Comment:
What is the point of grouping these two cases together and then always breaking in the case of `as_tensor` ?

Reply:
Maybe add that as_tensor has an overloaded schema and can either have a tensor or a list as the first input